### ISSUE TITLE
Expose dhParameters to string endpoints

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6924]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)| @hynek reported|
|-|-|
|Trac ID|trac#6924|
|Type|enhancement|
|Created|2014-01-19 22:45:05Z|
|Branch|https://github.com/twisted/twisted/tree/dhe-string-endpoints-6924|

We have DHE from [#6799](https://github.com/twisted/twisted/issues/6799), now let’s let people use it from string endpoints!

<details><summary>Searchable metadata</summary>

```
trac-id__6924 6924
type__enhancement enhancement
reporter__hynek hynek
priority__normal normal
milestone__ 
branch__branches_dhe_string_endpoints_6924 branches/dhe-string-endpoints-6924
branch_author__hynek hynek
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1390171505000000 1390171505000000
changetime__1390407252000000 1390407252000000
version__None None
owner__hynek hynek

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)<a name="note_1"></a>|@hynek commented|
|-|-|

(In [41380]) Branching to dhe-string-endpoints-6924.

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)<a name="note_2"></a>|@hynek commented|
|-|-|

Here we go. The code changes are trivial and the bots don’t disagree.

Unfortunately there’s some churn in endpoints.rst since whitespace fixes touched nearly every paragraph. That’s why I’ve split it up into [ just the fixups](https://github.com/twisted/twisted/commit/dc818984ab81a435493850e45e4a5802560a82c8) and [the actual docs update](https://github.com/twisted/twisted/commit/8cdae1133b1c10f7ce0d9403133fa1d5f8e7b02d). On the long run, I believe it’s worth it though.

**Comment by twisted-trac** [comment]
|<img alt="Alex's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|Alex set owner to @hynek|
|-|-|

This looks good to me, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)<a name="note_4"></a>|@hynek set status to `closed`|
|-|-|

(In [41413]) Merge dhe-string-endpoints-6924

Author: hynek
Reviewer: Alex
Fixes: [#6924](https://github.com/twisted/twisted/issues/6924)

SSL server endpoint string descriptions now support the specification of Diffie-Hellman key exchange parameter files.
